torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -1941,6 +1944,55 @@ std::vector<Val*> Index::getPerDimLogicalIndex(
   return getRootIndices(consumer_tv, loops, index_from_id_graph);
 }
 
+kir::TensorIndex* Index::getIndexForNonEqualDomains(
+    TensorView* producer_tv, 
+    TensorView* unknown_tv,
+    TensorView* consumer_tv,
+    const std::vector<kir::ForLoop*>& loops,
+    const std::unordered_map<IterDomain*, Val*>& override_index) {
+
+  const auto producer_indices = Index::getPerDimLogicalIndex(consumer_tv, loops);
+  const auto producer_strides = Index::getStrides(producer_tv);
+  const auto producer_domains = producer_tv->getRootDomain();
+  TORCH_INTERNAL_ASSERT(producer_indices.size() == producer_strides.size());
+  
+  Val* producer_flatten_index = nullptr;
+  for(size_t i = 0; i < producer_strides.size(); ++i) {
+      producer_flatten_index = SimplifyingIrBuilder::addExpr(
+        producer_flatten_index, 
+        SimplifyingIrBuilder::mulExpr(producer_indices[i], producer_strides[i])
+      );
+  }
+
+
+  const auto unknown_domains = unknown_tv->getRootDomain();
+  const auto unknown_strides = getStrides(unknown_tv);
+
+  std::vector<Val*> unknwon_index;
+  for(size_t dim = 0; dim < producer_strides.size(); ++dim) {
+    auto producer_d = producer_domains[dim];
+
+    auto override_it = override_index.find(unknown_domains[dim]);
+    const bool is_overriden = override_it != override_index.end();
+
+    if(is_overriden) {
+      unknwon_index.push_back(IrBuilder::mulExpr(override_it->second, unknown_strides[dim]));
+    }
+    else {
+      unknwon_index.push_back(IrBuilder::mulExpr(
+        IrBuilder::modExpr(
+          IrBuilder::divExpr(
+            producer_flatten_index, 
+            producer_strides[dim]),  
+          producer_d->extent() 
```

Comment:
I'm unclear what you're trying to do here. I guess `divExpr(producer_flatten_index, producer_strides[dim])` is to grab an index component from the flattened index val? If so, wouldn't it make more sense to do using `producer_indices`? It seems this part is just reversing what you've already done.

Reply:
You are right, I do a lot of unnecessary work here. And will fix quickly.